qiskit/providers/ibmq/job/ibmqjob.py

```diff
@@ -627,6 +627,9 @@ def share_level(self) -> str:
         Returns:
             The share level of the job.
         """
+        warnings.warn("`IBMQJob.share_level()` method is no longer supported "
+                      "and will be removed in a future release.",
+                      Warning, stacklevel=2)
         if not self._share_level and not self._refreshed:
             self.refresh()
         return self._share_level
```

Comment:
There is no need to call `self.refresh()` (which is a network call) anymore since share level will always be `None`.